### PR TITLE
fix regression where .Rproj.user folder not marked hidden

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/15514.

### Approach

The `computeUserDirectory()` now also creates the default project directory, which causes us to avoid an older code path where we also would have marked the project directory as hidden. This PR ensures we unconditionally mark the `.Rproj.user` directory as hidden on Windows, even if it was already created before.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/15514. Windows only.

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/session/projects/SessionProjectContext.cpp [hunk 1]
```diff
@@ -183,23 +183,21 @@ Error computeScratchPaths(const FilePath& projectFile,
...
    FilePath projectUserDir = computeUserDir(projectFile, projectConfig);
-   if (!projectUserDir.exists())
-   {
-      // create
-      Error error = projectUserDir.ensureDirectory();
-      if (error)
-         return error;
+   
+   // make sure it exists
+   Error error = projectUserDir.ensureDirectory();
```
Comment (kevinushey) [inline src/cpp/session/projects/SessionProjectContext.cpp:190]: `ensureDirectory()` already calls `exists()` internally, so it's unnecessary to call it here.

File: src/cpp/session/projects/SessionProjectContext.cpp [hunk 2]
```diff
@@ -183,23 +183,21 @@ Error computeScratchPaths(const FilePath& projectFile,
...
+   if (error)
+      return error;
 
-      // mark hidden if we are on win32
+   // mark hidden if we are on win32
 #ifdef _WIN32
-      error = core::system::makeFileHidden(projectUserDir);
-      if (error)
-         return error;
+   error = core::system::makeFileHidden(projectUserDir);
```
Comment (kevinushey) [inline src/cpp/session/projects/SessionProjectContext.cpp:196]: This also ensures any `.Rproj.user` directories which already exist but weren't marked as hidden now are.

**Review by gtritchie — CHANGES_REQUESTED** [review]
LGTM but aimed at wrong branch (main instead of 2024.12.1).

**Review by gtritchie — APPROVED** [review]
LGTM!

**Pull request merged**